### PR TITLE
[SKPR-551]: Restore the latest backup.

### PR DESCRIPTION
**Pull request by fubarhouse**

### CODE REVIEW

File: src/scripts/restore-latest.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+restore() {
+    if [ -z "${SKPR_BACKUP}" ]; then
+        SKPR_BACKUP=$(skpr backup list "${SKPR_ENV}" --json | jq .[0].Name)
```
Comment (nickschuch) [inline src/scripts/restore-latest.sh]: Check if you need raw response

File: src/scripts/restore-latest.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          echo "Failed to get name of most recent backup"
+          exit 1
+        fi
+        # shellcheck disable=SC2086
+        echo "export SKPR_VERSION=$SKPR_VERSION" >> $BASH_ENV
+    fi
+    echo "Deploying backup: ${SKPR_VERSION}"
+
+    # shellcheck disable=SC2086
+    skpr deploy ${SKPR_ENV} ${SKPR_BACKUP}
```
Comment (nickschuch) [inline src/scripts/restore-latest.sh]: Needs to be `skpr restore create`

File: src/commands/restore-latest.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+parameters:
+  env:
+    type: string
+    default: ""
+    description: "The app environment"
+
+steps:
+  - run:
+      environment:
+        SKPR_ENV: <<parameters.env>>
```
Comment (nickschuch) [inline src/commands/restore-latest.yml]: Needs source and target environment names

File: src/scripts/restore-latest.sh (hunk 1 above)
Comment (fubarhouse) [reply src/scripts/restore-latest.sh]: Checked and nope - doesn't support indexing when using raw.

File: src/scripts/restore-latest.sh (hunk 2 above)
Comment (fubarhouse) [reply src/scripts/restore-latest.sh]: https://github.com/skpr/skpr-orb/blob/8e5c8bf6cea3b2ecf433de0c4d53dd0d2e45f7f3/src/scripts/restore.sh#L19

File: src/commands/restore-latest.yml (hunk 3 above)
Comment (fubarhouse) [reply src/commands/restore-latest.yml]: https://github.com/skpr/skpr-orb/blob/8e5c8bf6cea3b2ecf433de0c4d53dd0d2e45f7f3/src/commands/restore.yml#L5-L12

File: src/scripts/restore.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+restore() {
+	if [ -z "${SKPR_BACKUP}" ]; then
+		SKPR_BACKUP=$(skpr backup list "${SKPR_SOURCE}" --json | jq -r '[.][0][0].Name')
+		if $? -gt 0; then
+			echo "Failed to get name of most recent backup"
+			exit 1
+		fi
+		# shellcheck disable=SC2086
+		echo "export SKPR_BACKUP=SKPR_BACKUP" >>$BASH_ENV
```
Comment (nickschuch) [inline src/scripts/restore.sh]: Can you explain what this is doing? Looks like it's setting `SKPR_BACKUP` to `SKPR_BACKUP` (the string)

Comment (fubarhouse) [reply src/scripts/restore.sh]: Fixed up!

https://github.com/skpr/skpr-orb/blob/53bba37e8237d3d0a9a485985493a6602c780bac/src/scripts/restore.sh#L9-L13

**Comment by fubarhouse** [comment]
Is there anything else we need to do here, or can we merge?

File: src/commands/restore.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    default: ""
+    description: "The source environment to grab the backup from"
+  target:
+    type: string
+    default: ""
+    description: "The target environment to restore the backup to"
+  backup:
+    type: string
+    default: ""
+    description: "If applicable, the backup name to restore"
```
Comment (nickschuch) [inline src/commands/restore.yml]: What does "If applicable mean?". What happens if you don't provide it?

File: src/scripts/restore.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+		if $? -gt 0; then
+			echo "Failed to get name of most recent backup"
+			exit 1
+		fi
+		# shellcheck disable=SC2086
+		echo "export SKPR_BACKUP=${SKPR_BACKUP}" >>$BASH_ENV
+	fi
+	echo "Restoring backup '${SKPR_BACKUP}' to ${SKPR_TARGET}"
+
+	if [ -z "${SKPR_SOURCE}" ] || [ -z "${SKPR_TARGET}" ]; then
```
Comment (nickschuch) [inline src/scripts/restore.sh]: Split this out to check one at a time?

Comment (nickschuch) [reply src/scripts/restore.sh]: The error bundles both together and might get confusing for an end user.

File: src/commands/restore.yml (hunk 5 above)
Comment (fubarhouse) [reply src/commands/restore.yml]: I will correct the sentence - however it will choose the latest available backup from the source if left empty.

Comment (fubarhouse) [reply src/commands/restore.yml]: I've updated the description to show what happens if left empty, vs what the parameter actually does.

File: src/scripts/restore.sh (hunk 6 above)
Comment (fubarhouse) [reply src/scripts/restore.sh]: Separated out now.

**Comment by nickschuch** [comment]
LGTM!

**Pull request merged**